pkg/manager/manager.go

```diff
@@ -277,6 +281,22 @@ type Options struct {
 	newPprofListener       func(addr string) (net.Listener, error)
 }
 
+// Hookable is implemented by Managers that support running hooks during
+// controller lifecycle phases.
+type Hookable interface {
+	// Hook allows to add Runnables as hooks to modify the behavior.
+	Hook(hook HookType, runnable Runnable) error
+}
```

Comment:
`manager.New` returns a `Manager` interface, which I think means the `Hook` function will be inaccessible without a type assertion.

Is that the intention?